integration/hive/hive-openlineage-hook/src/main/java/io/openlineage/hive/hooks/Faceting.java

```diff
@@ -212,24 +265,30 @@ private static ColumnLineageDatasetFacet getIdentityCLL(
                     .build())
         .forEach(f -> cldffb.put(f.getInputFields().get(0).getField(), f));
 
-    ExportWork work =
-        (ExportWork) olContext.getHookContext().getQueryPlan().getRootTasks().get(0).getWork();
     ColumnLineageDatasetFacetBuilder columnLineageDatasetFacetBuilder =
         ol.newColumnLineageDatasetFacetBuilder().fields(cldffb.build());
-    if (work.getTableSpec().getPartSpec() != null && !work.getTableSpec().getPartSpec().isEmpty()) {
-      List<InputField> collect1 =
-          fields.stream()
-              .filter(f -> work.getTableSpec().getPartSpec().containsKey(f.getName()))
-              .map(
-                  f ->
-                      getInputField(
-                          inputDataset,
-                          f,
-                          TransformationInfo.indirect(TransformationInfo.Subtypes.FILTER)))
-              .collect(Collectors.toList());
-      columnLineageDatasetFacetBuilder.dataset(collect1);
-    }
 
+    if (olContext.getHookContext().getQueryPlan().getOperation() == HiveOperation.EXPORT) {
+      // in case of EXPORT there is a possibility if filtering by using PARTITION keyword
+      // so we need to put the partitioning columns in the dataset dependencies with FILTER
+      // transformation
+      ExportWork work =
+          (ExportWork) olContext.getHookContext().getQueryPlan().getRootTasks().get(0).getWork();
+      Map<String, String> partSpec = work.getTableSpec().getPartSpec();
+      if (partSpec != null && !partSpec.isEmpty()) {
+        List<InputField> datasetDependencyInputs =
+            fields.stream()
+                .filter(f -> partSpec.containsKey(f.getName()))
+                .map(
+                    f ->
+                        getInputField(
+                            inputDataset,
+                            f,
+                            TransformationInfo.indirect(TransformationInfo.Subtypes.FILTER)))
```

Comment:
But this doesn't mean that input dataset even had this column in schema. Instead you can use datasetSubset facet to describe that output is partial

Reply:
but can you have a hive table that is partitioned without the partition column being mentioned in the schema?

btw in this PR I only moved it a bit, the code is already on main so we can just tackle it as separate issue there

Reply:
The *output* table is partitioned. But here we're adding field to *input* dataset, which may be not